src/main/java/com/teks/users/user/service/UserServiceImplementation.java

```diff
@@ -11,45 +20,85 @@
 @Service
 public class UserServiceImplementation implements UserService {
     private final UserRepository userRepository;
+    private final Utils utils;
 
-    public UserServiceImplementation(UserRepository userRepository){
+    public UserServiceImplementation(UserRepository userRepository,  Utils utils){
         this.userRepository = userRepository;
+        this.utils = utils;
     }
+//    public UserServiceImplementation(){
+//
+//    }
 
     @Override
-    public void createUser(User user){
-        userRepository.save(user);
+    public UserDto createUser(UserDto userDto){
+        User newUser = new User();
+        BeanUtils.copyProperties(userDto, newUser);
+
+        newUser.setEncryptedPassword("test");
+        newUser.setEmailVerificationStatus(utils.emailIsValid(userDto));
+        if (!newUser.getEmailVerificationStatus()){
+            throw new InvalidEmailException("Email Is Not Valid!");
```

Comment:
If you don't have an ExceptionsHandler class this doesn't do anything different than just a normal RuntimeException